codec-http/src/main/java/io/netty/handler/codec/http/HttpResponseStatus.java

```diff
@@ -447,7 +449,21 @@ public static HttpResponseStatus valueOf(int code) {
     }
 
     /**
-     * Parses the specified HTTP status line into a {@link HttpResponseStatus}.  The expected formats of the line are:
+     * Returns the {@link HttpResponseStatus} represented by the specified {@code code} and {@code reasonPhrase}.
+     * If the specified code is a standard HTTP status {@code code} and {@code reasonPhrase}, a cached instance
+     * will be returned. Otherwise, a new instance will be returned.
+     * @param code The response code value.
+     * @param reasonPhrase The response code reason phrase.
+     * @return the {@link HttpResponseStatus} represented by the specified {@code code} and {@code reasonPhrase}.
+     */
+    public static HttpResponseStatus valueOf(int code, String reasonPhrase) {
+        HttpResponseStatus responseStatus = valueOf(code);
+        return responseStatus.reasonPhrase().contentEquals(reasonPhrase) ? responseStatus :
```

Comment:
contentEquals -> equals

Reply:
I don't see any clear guidance as if it should be treated as case insensitive or not. can you point me to the RFC which defines the case sensitivity here? If possible I would like to avoid object allocation even if the case doesn't match up.

https://tools.ietf.org/html/rfc7230#section-3.1.2
> A client SHOULD ignore the reason-phrase content.
> reason-phrase  = *( HTAB / SP / VCHAR / obs-text )

Reply:
IMHO `contentEquals` is fine here.

Reply:
@Scottmitch I mean just, if we compare two string, it's better to use `equals` instead of `contentEquals`. This is not related with case sensitivity.